modules/pipeline/dbclient/op_pipeline.go

```diff
@@ -326,6 +326,7 @@ func (client *Client) PageListPipelines(req apistructs.PipelinePageListRequest,
 			if len(definitionReq.Name) > 0 {
 				baseSQL.Where(fmt.Sprintf("%v.name like ?", definitiondb.PipelineDefinition{}.TableName()), "%"+definitionReq.Name+"%")
 			}
+			baseSQL.Where(fmt.Sprintf("%s.location = ?", definitiondb.PipelineDefinition{}.TableName()), definitionReq.Location)
```

Comment:
I think you should check if location's length > 0 ?

Reply:
done